src/main/java/com/p6spy/engine/common/Value.java

```diff
@@ -117,43 +118,28 @@ public String convertToString(Object value) {
         result = value.toString();
       }
 
-      result = quoteIfNeeded(result, value);
+      result = quoteIfNeeded(result, value, binaryFormat);
```

Comment:
What about moving this logic into `BinaryFormat#toString(byte[])`? I think having self-sufficient strategy would be a good choice despite small code duplication.

Reply:
I modified the pull request so that BinaryFormat#toString wraps its return value in quotes if needed and convertToString returns early if a BinaryFormat is used.